mediabox.sh

```diff
@@ -125,30 +147,30 @@ printf "\n\n"
 printf "Configuring Deluge daemon access - UHTTPD index file - Permissions \n\n"
 
 # Configure DelugeVPN: Set Daemon access on, delete the core.conf~ file
-`while [ ! -f delugevpn/config/core.conf ]; do sleep 1; done`
-`docker stop delugevpn > /dev/null 2>&1`
-`rm delugevpn/config/core.conf~ > /dev/null 2>&1`
-`sed -i 's/"allow_remote": false,/"allow_remote": true,/g'  delugevpn/config/core.conf`
-`sed -i 's/"move_completed": false,/"move_completed": true,/g'  delugevpn/config/core.conf`
-`docker start delugevpn > /dev/null 2>&1`
+while [ ! -f "$datadirectory"/delugevpn/config/core.conf ]; do sleep 1; done
+docker stop delugevpn > /dev/null 2>&1
+rm -f "$datadirectory"/delugevpn/config/core.conf~ > /dev/null 2>&1
+perl -i -pe 's/"allow_remote": false,/"allow_remote": true,/g'  "$datadirectory"/delugevpn/config/core.conf
+perl -i -pe 's/"move_completed": false,/"move_completed": true,/g'  "$datadirectory"/delugevpn/config/core.conf
+docker start delugevpn > /dev/null 2>&1
 
 # Push the Deluge Daemon Access info the to Auth file
-`echo $daemonun:$daemonpass:10 >> ./delugevpn/config/auth`
+echo $daemonun:$daemonpass:10 >> "$datadirectory"/delugevpn/config/auth
 
 # Configure UHTTPD settings and Index file
-`docker stop uhttpd > /dev/null 2>&1`
-`mv index.html www/index.html` 
-`sed -i "s/locip/$locip/g" www/index.html`
-`sed -i "s/daemonun/$daemonun/g" www/index.html`
-`sed -i "s/daemonpass/$daemonpass/g" www/index.html`
-`cp .env www/env.txt`
-`docker start uhttpd > /dev/null 2>&1`
+docker stop uhttpd > /dev/null 2>&1
+cp index.html "$datadirectory"/www/index.html
+perl -i -pe "s/locip/$locip/g" "$datadirectory"/www/index.html
+perl -i -pe "s/daemonun/$daemonun/g" "$datadirectory"/www/index.html
+perl -i -pe "s/daemonpass/$daemonpass/g" "$datadirectory"/www/index.html
+cp .env "$datadirectory"/www/env.txt
+docker start uhttpd > /dev/null 2>&1
 
 # Fix the Healthcheck in Minio
 docker exec minio sed -i "s/404/403/g" /usr/bin/healthcheck.sh
```

Comment:
If `perl -i -pe` is like a "drop-in" replacement for `sed -i` should we use that here too?